addons/mail_client_extension/controllers/main.py

```diff
@@ -243,9 +273,50 @@ def res_partner_create(self, email, name, company, **kwargs):
             'name': name,
             'email': email,
         }
-        if company > -1:
+        if company and company > -1:
             partner_info['parent_id'] = company
         partner = request.env['res.partner'].create(partner_info)
 
         response = {'id': partner.id}
         return response
+
+    @http.route('/mail_client_extension/log_mail_content', type="json", auth="outlook", cors="*")
+    def log_mail_content(self, model, res_id, message):
+        assert model in self._get_loggable_models()
+        record = request.env[model].browse(res_id)
+        record.message_post(body=message)
+
+    def _get_loggable_models(self):
+        return ['res.partner']
+
+    @http.route('/mail_client_extension/partner/enrich_company', type="json", auth="outlook", cors="*")
+    def res_partner_enrich_company(self, partner_id):
+        """
+        Route used when the user clicks on the enrich partner button
+        it will first try to find a related company in the database if none are found,
+        it will enrich using IAP
+        """
+        response = {}
+
+        partner = request.env['res.partner'].browse(partner_id)
+
+        normalized_email = tools.email_normalize(partner.email)
+        if not normalized_email:
+            response['error'] = 'Bad email.'
+            return response
+        company_domain = normalized_email.split('@')[1]
+
+        company, enrichment_info = self._create_company_from_iap(company_domain)
```

Comment:
So we also create the company? It's not only enrichment?
Should be made clear in the method name / docstring then.

Reply:
That is correct, a new company is created when IAP finds a company.

The method name has been changed accordingly, a docstring has also been added to explain this.